fvm/evm/offchain/query/view.go

```diff
@@ -0,0 +1,371 @@
+package query
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/holiman/uint256"
+	gethCommon "github.com/onflow/go-ethereum/common"
+	gethTypes "github.com/onflow/go-ethereum/core/types"
+	gethCrypto "github.com/onflow/go-ethereum/crypto"
+	gethTracers "github.com/onflow/go-ethereum/eth/tracers"
+
+	"github.com/onflow/flow-go/fvm/evm/emulator"
+	"github.com/onflow/flow-go/fvm/evm/emulator/state"
+	"github.com/onflow/flow-go/fvm/evm/offchain/blocks"
+	"github.com/onflow/flow-go/fvm/evm/offchain/storage"
+	"github.com/onflow/flow-go/fvm/evm/offchain/sync"
+	"github.com/onflow/flow-go/fvm/evm/types"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// View provides query capabilities over
+// an specific state of the EVM chain.
+type View struct {
+	chainID         flow.ChainID
+	rootAddr        flow.Address
+	storage         *storage.EphemeralStorage
+	tracer          *gethTracers.Tracer
+	extraPCs        []types.PrecompiledContract
+	maxCallGasLimit uint64
+}
+
+// NewView constructs a new view.
+func NewView(
+	chainID flow.ChainID,
+	rootAddr flow.Address,
+	storage *storage.EphemeralStorage,
+	maxCallGasLimit uint64,
+) *View {
+	return &View{
+		chainID:         chainID,
+		rootAddr:        rootAddr,
+		storage:         storage,
+		maxCallGasLimit: maxCallGasLimit,
+	}
+}
+
+// GetBlockMeta return block meta data
+func (v *View) GetBlockMeta() (*blocks.Meta, error) {
```

Comment:
Maybe `GetLatestBlockMeta` is a more fitting name for this method.

Reply:
I removed this, as it was not needed, I added the logic to pass the block context provider.